solutions/USACO jan19-mountain-view.mdx

```diff
@@ -0,0 +1,75 @@
+---
+id: usaco-896
+source: USACO Silver 2019 January
+title: Mountain View
+author: Jesse Choe
+---
+
+[Official Analysis](http://usaco.org/current/data/sol_prob3_silver_dec20.html)
+
+**Time Complexity:** $\mathcal{O}(n\log n)$
+
+<LanguageSection>
+<CPPSection>
+
+## Sorting Struct Using Comparator Function
+
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+
+#define FastIO ios_base::sync_with_stdio(0); cin.tie(0)
+#define pb push_back
+#define all(x) begin(x), end(x)
+#define sor(x) sort(all(x))
+#define ins insert
+#define f first
+#define s second
+#define sz(x) (int)(x).size()
+#define len(x) (int)(x).length()
+#define FOR(i, a, b) for(int i=a; i<b; ++i)
+#define ROF(i, b, a) for(int i=(b)-1; i>=a; --i)
+
+void setIO(string fileName = ""){
+	FastIO;
+	if(len(fileName)){
+		freopen((fileName+".in").c_str(), "r", stdin);
+		freopen((fileName+".out").c_str(), "w", stdout);
+	}
+}
+
+struct Mountain {
+    int l, r;
+};
+
+bool cmp(Mountain a, Mountain b){
+    if(a.l == b.l){
+        return a.r > b.r;
+    }
+    return a.l < b.l;
+}
```

Comment:
then u wouldnt have this

Reply:
Do you explicitly have to sort `r` in decreasing order? In this case, it's possible to justify using custom comparator instead of negating it.

Reply:
hmm thats mb then